core/src/main/java/nl/nn/adapterframework/filesystem/ExchangeFileSystem.java

```diff
@@ -1002,12 +1002,18 @@ protected Date getDateTimeReceived(EmailMessage emailMessage) {
 
 	@Override
 	public String getSubject(EmailMessage emailMessage) throws FileSystemException {
+		ItemId id=null;
 		try {
-			if (emailMessage.getId()!=null) { // attachments don't have an id, but appear to be loaded at the same time as the main message
+			id = emailMessage.getId();
+			if (id!=null) { // attachments don't have an id, but appear to be loaded at the same time as the main message
 				emailMessage.load(new PropertySet(ItemSchema.Subject));
 			}
 			return emailMessage.getSubject();
 		} catch (Exception e) {
+			if (id==null) {
+				log.warn("Could not get Subject for message without ItemId", e);
+				return null;
```

Comment:
Ik zou verwachten dat als er geen ID is hij daar returned. Niet in de exceptie...?

Reply:
Als er geen Id is kan er wel een subject zijn, dat ook wel wel gevonden kan worden.